docs/configuration.md

```diff
@@ -0,0 +1,162 @@
+- [Tekton Pipelines](#tekton-pipelines)
+- [Tekton Tasks](#tekton-tasks)
+- [Feature Gates](#feature-gates)
+
+## SSP Custom Resource (CR)
+
+To activate the operator, create the SSP Custom Resource (CR):
+```
+apiVersion: ssp.kubevirt.io/v1beta2
+kind: SSP
+metadata:
+  annotations:
+    ssp.kubevirt.io/vm-console-proxy-enabled: "true"
+    ssp.kubevirt.io/vm-console-proxy-namespace: "kubevirt"
+  name: ssp-sample
+  namespace: kubevirt
+spec:
+  commonTemplates:
+    namespace: kubevirt
+  templateValidator:
+    replicas: 2
+  featureGates:
+    deployTektonTaskResources: true
+  tektonPipelines:
+    namespace: kubevirt
+  tektonTasks:
+    namespace: kubevirt
+```
+
+## Annotations
+
+### VM Console Proxy
+
+These annotations are used by VM console proxy operand.
+
+```
+apiVersion: ssp.kubevirt.io/v1beta1
+kind: SSP
+metadata:
+  annotations:
+    ssp.kubevirt.io/vm-console-proxy-enabled: "true" # If not set, then by default is false
+    ssp.kubevirt.io/vm-console-proxy-namespace: "kubevirt" # If not set, then default namespace is "kubevirt"
+  name: ssp-sample
+  namespace: kubevirt
+spec: {}
+```
+
+### Pause Operator
+
+This annotation will pause operator reconciliation.
+
+```
+apiVersion: ssp.kubevirt.io/v1beta1
+kind: SSP
+metadata:
+  annotations:
+    kubevirt.io/operator.paused: "false" # If not set, then by default is false
+  name: ssp-sample
+  namespace: kubevirt
+spec: {}
+```
+
+The operator will not react to any changes to the `SSP` resource
+or any of the watched resources. If a paused `SSP` resource is deleted,
+the operator will still cleanup all the dependent resources.
+
+## Common Templates
+
+A set of common templates to create KubeVirt Virtual Machines (VMs).
+
+```
+apiVersion: ssp.kubevirt.io/v1beta1
+kind: SSP
+metadata:
+  name: ssp-sample
+  namespace: kubevirt
+spec:
+  commonTemplates:
+    namespace: kubevirt
+```
+
+## Template Validator
+
+Validator is designed to inspect virtual machines (VMs) and detect any violations of the rules established within templates.
+
+```
+apiVersion: ssp.kubevirt.io/v1beta1
+kind: SSP
+metadata:
+  name: ssp-sample
+  namespace: kubevirt
+spec:
+  templateValidator:
+    replicas: 2 # Customize the number of replicas for the validator deployment
+```
+
+## Tekton Pipelines
+
+Define the namespace of Tekton Pipelines, where it will be deployed.
+
+```
+apiVersion: ssp.kubevirt.io/v1beta2
+kind: SSP
+metadata:
+  name: ssp-sample
+  namespace: kubevirt
+spec:
+  tektonPipelines:
+    namespace: kubevirt
+```
+
+## Tekton Tasks
+
+Define the namespace of Tekton Tasks, where it will be deployed.
+
+```
+apiVersion: ssp.kubevirt.io/v1beta2
+kind: SSP
+metadata:
+  name: ssp-sample
+  namespace: kubevirt
+spec:
+  tektonTasks:
+    namespace: kubevirt
+```
+
+## Feature Gates
+
+The featureGates field is an optional set of optional boolean feature enabler.
+The features in the list are advanced or new features that are not enabled by default.
```

Comment:
You could call them experimental features?

Reply:
In our case, yes, in SSP we have only experimental features.